instrumentation/jvm-metrics/src/main/java/com/splunk/opentelemetry/instrumentation/jvmmetrics/AllocatedMemoryMetrics.java

```diff
@@ -42,12 +39,7 @@ public void bindTo(MeterRegistry registry) {
     }
 
     AllocationTracker allocationTracker = new AllocationTracker();
-    FunctionCounter.builder(
-            METRIC_NAME, allocationTracker, AllocationTracker::getCumulativeAllocationTotal)
-        .description("Approximate sum of heap allocations")
-        .baseUnit(BaseUnits.BYTES)
-        .tag("type", "heap")
-        .register(registry);
+    meterFactory.accept(allocationTracker, allocationTracker::getCumulativeAllocationTotal);
```

Comment:
This feels unnecessarily callbacky to me now. Why not get rid of `install` completely and replace it with a public method that just delegates to `allocationTracker.getCumulativeAllocationTotal()`? With that approach, the two implementations can just call that directly, rather than going thru an extra `Supplier`?

Reply:
I went with callbacks specially for you :) An alternative would be to use inheritance and use this class as a super class and implement registering the meter in a child class. Would you like that more?

Reply:
The problem is that the `install` method also does some other things besides invoking callback. So it really would look more like
```java
public void install() {
  AllocatedMemoryMetrics metrics = new AllocatedMemoryMetrics();
  if (!metrics.canEnable()) {
    return;
  }
  createMeter();
  metrics.registerListener(); // only for gc metrics
}
```
so there will be a bit of duplication. Is this how you'd like it?

Reply:
Right, it also contains those checks to see if we even _can_ do the memory metrics stuff. This might be highlighting the fact that the class has more than one responsibility.

I think this duplication isn't egregious and does make things slightly cleaner.

Reply:
changed as requested